### ISSUE TITLE
ISSUE: Edit department is rolled-back

### ISSUE THREAD

**Issue by SelimHorri** [SelimHorri/project-tracking-system-android-app#48]
When hitting the '**edit department**' button,  _javax.persistance.RollbackException_ comes in saying 
"**_could not commit JPA transaction_**"

![edi_dept_eeror](https://user-images.githubusercontent.com/40203497/124652272-d6c92f80-de93-11eb-98a9-44408edc1a93.png)

**Comment by SelimHorri** [comment]
those are logs generated from server-side:

![edit_dept_prob11](https://user-images.githubusercontent.com/40203497/124653043-bbaaef80-de94-11eb-9e16-38fdc0a8d373.png)

The problem should be from mobile client-side in department service implementation update method : department.getLocation()

**Comment by BadreIdoudi** [comment]
![image](https://user-images.githubusercontent.com/23629822/124793346-cd000480-df45-11eb-850d-f7fc94a137f8.png)
it add a new location !, could u verify plz

**Comment by BadreIdoudi** [comment]
![image](https://user-images.githubusercontent.com/23629822/124793501-f751c200-df45-11eb-95d7-7ba4dcce9565.png)
do u think that the issue si there ?

**Comment by SelimHorri** [comment]
yeah @BadreIdoudi definitely, i need to find another consistent way to pass data in request body...

**Comment by SelimHorri** [comment]
Defining a new structure of the Map variable may resolve issue with **department.getLocation()** like below:
Map<String, ? extends Object>
